### PR TITLE
Update password comparison to timing safe

### PR DESCRIPTION
**Pull request by BlenderDude**
## :recycle: Current situation

Current password check code allows for the hash to be guessed utilizing a timing attack.

## :bulb: Proposed solution

Simple update to the password check function that utilizes `crypto.timingSafeEqual` such that comparisons are always the same time.

## :gear: Release Notes

Increased password security with timing safe comparison

### Testing

Testing should be covered by existing cases in `auth.e2e-spec.ts`

### CODE REVIEW

**Comment by bwp91** [comment]
Hi @BlenderDude 
Thanks for your time on this PR.

Can you explain the changes in laymans terms?

I guess this would not result in any difference in behaviour for a standard user logging in?

**Review by donavanbecker — APPROVED** [review]
